deepeval/synthesizer/synthesizer.py

```diff
@@ -155,49 +154,68 @@ def generate_goldens(
-                         golden = Golden(input=evolved_input, context=context, source_file=source_file)
-                         goldens.append(golden)
+                        evolved_input = self._evolve_text(
+                            data.input,
+                            context=context,
+                            num_evolutions=num_evolutions,
+                            enable_breadth_evolve=enable_breadth_evolve,
+                        )
+                        source_file = (
+                            source_files[i]
+                            if source_files is not None
+                            else None
+                        )
+                        golden = Golden(
+                            input=evolved_input,
+                            context=context,
+                            source_file=source_file,
+                        )
+                        goldens.append(golden)
 
             self.synthetic_goldens.extend(goldens)
 
             return goldens
 
     # TODO
     def generate_goldens_from_docs(
-            self, 
-            paths: List[str],
-            num_context:int,
-            chunk_size:int=1024, 
-            chunk_overlap:int=0,
-            max_goldens_per_context: int = 2, 
-            num_evolutions: int = 1,
-            enable_breadth_evolve: bool = False
-            ):
-        
-        contexts = None
-        if self.multithreading:
-            cg = ContextGenerator(paths, 
-                                  chunk_size, 
-                                  chunk_overlap, 
-                                  multithreading=True)
-            contexts, source_files = cg.generate_contexts(num_context=num_context)
-        else:
-            cg = ContextGenerator(paths, 
-                                  chunk_size, 
-                                  chunk_overlap, 
-                                  multithreading=False),
-            contexts, source_files = cg.generate_contexts(num_context=num_context)
-        goldens = self.generate_goldens(contexts, 
-                                        max_goldens_per_context, 
-                                        num_evolutions, 
-                                        enable_breadth_evolve,
-                                        source_files)
-        return goldens
+        self,
+        docuemnt_paths: List[str],
+        max_goldens_per_document: int = 5,
+        chunk_size: int = 1024,
+        chunk_overlap: int = 0,
+        num_evolutions: int = 1,
+        enable_breadth_evolve: bool = False,
+    ):
+
+        if self.context_generator is None:
+            print("context generator")
+            self.context_generator = ContextGenerator(
+                docuemnt_paths,
+                chunk_size,
+                chunk_overlap,
+                multithreading=self.multithreading,
+            )
+
+        max_goldens_per_context = 2
```

Comment:
The default max goldens per context for doc goldens generation will be 2 for now. 

I first check if max goldens per doc < max goldens per context, which if it is, i should only generate 1 goldens per context.

Afterwards, I also divide max goldens per doc with max goldens per context to make sure num_context doesn't exceed max goldens per doc